### PR TITLE
rocmPackages.hsa-amd-aqlprofile-bin: 5.7.0 -> 5.7.1

### PR DESCRIPTION
**Pull request by Madouura**
Tracking: #197885

Made a nice little update script for this finally.
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/rocm-modules/5/hsa-amd-aqlprofile-bin/default.nix [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
 }:
 
 let
-  prefix = "hsa-amd-aqlprofile";
-  version = "5.7.0";
-  major = lib.versions.major version;
-  minor = lib.versions.minor version;
-  patch = lib.versions.patch version;
-  magic = lib.strings.concatStrings (lib.strings.intersperse "0" (lib.versions.splitVersion version));
+  updateScript = callPackage ./update.nix { };
```
Comment (SuperSandro2000) [inline pkgs/development/rocm-modules/5/hsa-amd-aqlprofile-bin/default.nix:9]:
```suggestion
```

File: pkgs/development/rocm-modules/5/hsa-amd-aqlprofile-bin/default.nix [hunk 2]
```diff
@@ -29,11 +25,13 @@ in stdenv.mkDerivation (finalAttrs: {
     runHook preInstall
 
     mkdir -p $out
-    cp -a opt/rocm-${version}/* $out
+    cp -a opt/rocm-${finalAttrs.version}/* $out
 
     runHook postInstall
   '';
 
+  passthru.updateScript = updateScript { inherit (finalAttrs) version; };
```
Comment (SuperSandro2000) [inline pkgs/development/rocm-modules/5/hsa-amd-aqlprofile-bin/default.nix:33]:
```suggestion
  passthru.updateScript = (callPackage ./update.nix { }) { inherit (finalAttrs) version; };
```

Comment (Madouura) [reply pkgs/development/rocm-modules/5/hsa-amd-aqlprofile-bin/default.nix]: Weird. Tried that before and it didn't work, but does now.

**Comment by pbsds** [comment]
Tried the update script via `nix-update -u ...`, and it works. The ELF binaries in `results/rocmPackages.hsa-amd-aqlprofile-bin/lib/` however are not marked executable.

**Comment by Madouura** [comment]
Didn't notice that, thank you for noting it!

**Review by pbsds — APPROVED** [review]

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 262750` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>rocmPackages.hsa-amd-aqlprofile-bin (rocmPackages_5.hsa-amd-aqlprofile-bin)</li>
    <li>rocmPackages.rocprofiler (rocmPackages_5.rocprofiler)</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Review by Flakebi — APPROVED** [review]

**Pull request merged**